src/client/clientmap.cpp

```diff
@@ -677,12 +677,24 @@ void ClientMap::renderPostFx(CameraMode cam_mode)
 	MapNode n = getNode(floatToInt(m_camera_position, BS));
 
 	const ContentFeatures& features = m_nodedef->get(n);
+
+	// Account for the near clipping plane to counter see-through "glitches":
+	// Check all nodes within a radius of near_plane (Manhattan distance)
+	bool any_nearby_solid = false;
+	f32 near_plane = m_client->getCamera()->getCameraNode()->getNearValue();
+	for (s8 x = -1; x <= 1; x += 2) {
+		for (s8 y = -1; y <= 1; y += 2) {
+			for (s8 z = -1; z <= 1; z += 2) {
```

Comment:
This is a cartesian product ${-1, 1}^3$. Should I more compactly (and possibly more efficiently?) golf this to a single for loop using some bitpacking?

Reply:
We have `g_6dirs` for this, also you can skip this loop if `allow_noclip`.

Reply:
These are actually 8 dirs (the cuboid corners rather than the face centers).

Reply:
oh huh
add it local to this file I guess, someone will hopefully move it to the right place if it's needed a second time